components/sync/driver/brave_sync_auth_manager.cc

```diff
@@ -21,14 +21,16 @@ std::string AppendBraveServiceKeyHeaderString() {
 }  // namespace
 
 BraveSyncAuthManager::BraveSyncAuthManager(
-    signin::IdentityManager* identity_manager,
+    signin::BraveIdentityManager* identity_manager,
     const AccountStateChangedCallback& account_state_changed,
     const CredentialsChangedCallback& credentials_changed)
-    : SyncAuthManager(identity_manager,
+    : SyncAuthManager(nullptr,
```

Comment:
This is dangerous, and any upstream change to `SyncAuthManager` that access identity_manager_ can easily cause runtime crash

Reply:
Redone completely in 4eff8e52e600de280c32a4bcbf7e0ff2a752f546 .